RiotNSE/NotificationService.swift

```diff
@@ -600,6 +613,106 @@ class NotificationService: UNNotificationServiceExtension {
+                    senderImage: senderImage)
+            intent.setImage(senderImage, forParameterNamed: \.sender)
+            do {
+                // TODO: figure out how to set _mentionsCurrentUser and _replyToCurrentUser in the context
+                let notificationContent = try notificationContent.updating(from: intent)
+                MXLog.debug("[NotificationService] makeCommunicationNotification: Calling onComplete.")
+                onComplete(notificationContent)
+            } catch {
+                MXLog.debug("[NotificationService] makeCommunicationNotification: error (roomMessage): \(error)")
+                onComplete(notificationContent)
+            }
+        default:
+            onComplete(notificationContent)
+        }
+    }
+
+
+    @available(iOS 15, *)
+    private func getMessageIntent(forEvent event: MXEvent,
+                                  body: String,
+                                  groupName: String?,
+                                  senderName: String,
+                                  senderImage: INImage?) -> INSendMessageIntent {
+        let intentType = getMessageIntentType(event.content["msgtype"] as? String, event)
+
+        let speakableGroupName = groupName.flatMap({ INSpeakableString(spokenPhrase: $0 )})
+
+        let sender = INPerson(
+            personHandle: INPersonHandle(value: event.sender, type: .unknown),
+            nameComponents: nil,
+            displayName: senderName,
+            image: senderImage,
+            contactIdentifier: nil,
+            customIdentifier: event.sender,
+            isMe: false,
+            suggestionType: .instantMessageAddress
+        )
+
+        let incomingMessageIntent = INSendMessageIntent(
+            // TODO: check
+            recipients: [],
+            outgoingMessageType: intentType,
+            content: body,
+            speakableGroupName: speakableGroupName,
+            conversationIdentifier: event.roomId,
+            // TODO: check
+            serviceName: nil,
+            sender: sender,
+            attachments: nil
+        )
+
+        let intent = INInteraction(intent: incomingMessageIntent, response: nil)
+        intent.direction = .incoming
+
+        intent.donate(completion: nil)
+
+        return incomingMessageIntent
+    }
+
+    @available(iOS 14, *)
+    private func getMessageIntentType(_ msgType: String?, _ event: MXEvent) -> INOutgoingMessageType {
+        guard let msgType = msgType else {
+            return .unknown
+        }
+
+        switch msgType {
+        case kMXMessageTypeEmote:
+            fallthrough
+        case kMXMessageTypeImage:
+            fallthrough
+        case kMXMessageTypeVideo:
+            fallthrough
+        case kMXMessageTypeFile:
+            return .unknown
+        case kMXMessageTypeAudio:
+            if event.isVoiceMessage() {
+                return .outgoingMessageAudio
+            }
+            return .unknown
```

Comment:
The docs for `.outgoingMessageAudio` say "An audio recording". Could this cover all audio messages?

Reply:
Probably, will change it